tdrs-backend/tdpservice/search_indexes/parsers/preparser.py

```diff
@@ -0,0 +1,262 @@
+            'year':         row[6:10],
+            'quarter':      row[10:11],
+            'type':         row[11:12],
+            'state_fips':   row[12:14],
+            'tribe_code':   row[14:17],
+            'program_type': row[17:20],
+            'edit':         row[20:21],
+            'encryption':   row[21:22],
+            'update':       row[22:23],
+        }
+
+        for key, value in header.items():
+            logger.debug('Header key %s: "%s"' % (key, value))
+
+        # TODO: Will need to be saved in parserLog, #1354
+
+        try:
+            # logger.debug("Given section: '%s'\t Header section: '%s'", given_section, section_map[header['type']])
+            # logger.debug("Given program type: '%s'\t Header program type: '%s'", data_type, header['program_type'])
+
+            if given_section != section_map[header['type']]:
+                raise ValueError('Given section does not match header section.')
+            if (data_type == 'TANF' and header['program_type'] != 'TAN')\
+                    or (data_type == 'SSP' and header['program_type'] != 'SSP')\
+                    or (data_type not in ['TANF', 'SSP']):
+                raise ValueError("Given data type does not match header program type.")
+        except KeyError as e:
+            logger.error('Ran into issue with header type: %s', e)
+
+        # TODO: could import schema from a schemas folder/file, would be reusable for other sections
+        # ugh, seems like we're limited to string for all values otherwise we have to cast to int in header and it fails
+        # with a raised ValueError instead of in the validator.errors
+        header_schema = {
+            'title':        {'type': 'string', 'required': True, 'allowed': ['HEADER']},
+            'year':         {'type': 'string', 'required': True, 'regex': '^20[0-9]{2}$'},
+            'quarter':      {'type': 'string', 'required': True, 'allowed': ['1', '2', '3', '4']},
+            'type':         {'type': 'string', 'required': True, 'allowed': ['A', 'C', 'G', 'S']},
+            'state_fips':   {'type': 'string', 'required': True, 'regex': '^[0-9]{2}$'},
+            'tribe_code':   {'type': 'string', 'required': False, 'regex': '^([0-9]{3}|[ ]{3})$'},
+            'program_type': {'type': 'string', 'required': True, 'allowed': ['TAN', 'SSP']},
+            'edit':         {'type': 'string', 'required': True, 'allowed': ['1', '2']},
+            'encryption':   {'type': 'string', 'required': True, 'allowed': ['E', ' ']},
+            'update':       {'type': 'string', 'required': True, 'allowed': ['N', 'D', 'U']},
+        }
+
+        validator = Validator(header_schema)
+        is_valid = validator.validate(header)
+        logger.debug(validator.errors)
+
+        return is_valid, validator
+
+    except KeyError as e:
+        logger.error('Exception validating header row, please see row and error.')
+        logger.error(row)
+        logger.error(e)
```

Comment:
Should these logs be more explanative?

Reply:
Multiple paths could trigger this exception but not from a result of bad data -- this was like a meta-programming thing around using cerberus incorrectly when I was learning it. Tbh, I think this might be better being removed.